extensions/amp-next-page/0.1/amp-next-page.js

```diff
@@ -72,38 +64,45 @@ export class AmpNextPage extends AMP.BaseElement {
       return;
     }
 
-    const {element} = this;
-    element.classList.add('i-amphtml-next-page');
-
-    const src = element.getAttribute('src');
-    if (src) {
-      return this.fetchConfig_().then(
-          config => this.register_(config, separator),
-          error => user().error(TAG, 'error fetching config', error));
-    } else {
-      const scriptElements = childElementsByTag(element, 'SCRIPT');
-      user().assert(scriptElements.length === 1,
-          `${TAG} should contain only one <script> child, or a URL specified `
-          + 'in [src]');
-      const scriptElement = scriptElements[0];
-      user().assert(isJsonScriptTag(scriptElement),
-          `${TAG} config should ` +
-          'be inside a <script> tag with type="application/json"');
-      const configJson = tryParseJson(scriptElement.textContent, error => {
-        user().error(TAG, 'failed to parse config', error);
-      });
-      this.register_(configJson, separator);
-    }
+    getServicePromiseForDoc(this.getAmpDoc(), SERVICE_ID).then(service => {
```

Comment:
Please `return` the promise to eliminate any possibility of a race between `buildCallback` and `layoutCallback` (it is a little know fact that `buildCallback` can return a `promise`)

Reply:
Oh cool! Done.